hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metrics/prometheus/PushGatewayMetricsReporter.java

```diff
@@ -77,4 +83,11 @@ private String getJobName() {
     }
     return configuredJobName;
   }
+
+  private static Map<String, String> parseLabels(String labels) {
+    return Pattern.compile("\\s*,\\s*")
+        .splitAsStream(labels.trim())
+        .map(s -> s.split(":", 2))
```

Comment:
What if labels are not in the format key:value? Should we add a validation or handle this scenario?

Reply:
Currently the way it behaves, it will convert input like 
```
"hudi:prometheus" -> "hudi" and "prometheus"
"hudi:prome:theus" -> "hudi" and "prome:theus"
"hudiprometheus" -> "hudiprometheus" and ""
```
This looks fine to me or do we want more stricter rules here?

Reply:
Ok. Can you add a UT for multiple key value pairs as well?

Reply:
Added in latest commit